v5.10 Changelog (#2604)
mattermost/docs: source/administration/changelog.md

Also see [changelog in progress](http://bit.ly/2nK3cVf) for the next release.

## Release v5.10

Release Date 2019-04-16

### Highlights

#### Infinite One Directional Scrolling
 - Removed the “Load more messages” button at the top of the screen to make it easier to read messages.
 
#### Interactive Ephemeral Messages

>  - Added ability to modify message attachments, allowing for richer integrations.

Reviewer: ```suggestion
 - Added support for [message actions and buttons](https://docs.mattermost.com/developer/interactive-messages.html) in ephemeral messages.
```

#### Configuration in Database

>  - Added support for managing configuration options directly in the database. This option honors the current ``config.json`` to maintain backwards compatibility while providing a way to easily manage configuration settings from the system console in more complex environments.

Reviewer 2: Propose
```suggestion
 - Added experimental support for storing ``config.json`` in the database, improving the system console experience on read-only filesystems. Storing the configuration in the database is optional, as the existing ``config.json`` remains fully supported.
```

 - Created a plugin component to override file previews.
 - Added support for plugins to create link tooltips.
 
#### Bulk Import/Export
 - Added User Preference fields in bulk export.
 - Added ability to include direct and group message channels and their posts in bulk export.
 - Added ability to include deactivated users in bulk import.
 
#### CLI
 - Created CLI command ``command show`` to allow seeing detailed information of a slash command.
 - Created CLI command ``webhook show`` to allow seeing detailed information of a webhook.
 - Created CLI command ``team rename`` to allow renaming teams.
 - Created CLI command ``channel search`` to allow searching for channels.
 
#### Administration
 - Improved default session timeout behavour, including changing the default ``SessionLengthWebInDays`` from 30 to 180 days.
 - Added full text search to the system console panel to easily find options in the configuration.
 - (Advanced Permissions) Split managing emoji permissions into "create", "delete own" and "delete others".
 - (Advanced Permissions) Added ``List_Public_Teams``, ``Join_Public_Teams``, ``List_Private_Teams`` and ``Join_Private_Teams`` permissions.

>  - Updated Mattermost to default to console logging in a human readable format.

Reviewer 2: This only affects developer mode, and not production: should we omit it from the release notes or qualify it as such?

Author: Removed,

 - Added support for LDAP groups search.
 - Added a setting to the system console to change the minimum length of hashtags.
 - Added support for setting Reply-To header in outbound Mattermost emails.
 - Added support for invalidating all email invitations from the system console.

### Bug Fixes
 - Fixed an issue where enterprise features became immediately unavailable when the enterprise license expired with a 15 day grace period.
 - Fixed an issue where a channel did not get removed from the unreads section if the user navigated out of it via a permalink.
 - Fixed an issue where a link from Access Control Groups to Group Filter on AD/LDAP did not work for subpath Site URL.
 - Fixed an issue where expired channels appeared in "My Channels" section of channel switcher if using the **Automatically Close Direct Messages** setting.
 - Fixed an issue where the text box reverted to default size after a user returned from the Integrations page.
 - Fixed an issue where the profile popover wasn't allowed to close itself when opened through an at-mention.
 - Fixed an issue where filtering by first name with Korean characters no longer worked for at-mentions.
 
### Compatibility

>  - Deprecated configurable ``timezones.json`` in favour of the existing hard-coded list built into the server. **XXXX NEEDS MORE DETAILS**

Author: Is this a deprecated feature or just part of the Configuration in Database?

https://mattermost.atlassian.net/browse/MM-14260

Reviewer 3: @lieut-data please comment here

Reviewer 2: I think the description (minus the `XXXXX` part ;P)) is sufficient: it's independent of the configuration in the database changes.

### config.json

Multiple setting options were added to `config.json`. Below is a list of the additions and their default values on install. The settings can be modified in `config.json`, or the System Console when available.
 
#### Changes to Team Edition and Enterprise Edition:

- Under ``"ExperimentalSettings":`` in ``config.json``:

>    - Added ``"RestrictSystemAdmin": false``, to 

Reviewer 2: ```suggestion
   - Added ``"RestrictSystemAdmin": false``, to optionally constrain even system admins from changing critical settings.
```

- Under ``"ServiceSettings":`` in ``config.json``:
   - Added ``"MinimumHashtagLength": 3``, to add the ability to change the minimum length of hashtags.
- Under ``"ElasticsearchSettings":`` in ``config.json``: 
   - Added ``"EnableAutocomplete": false``, ``"ChannelIndexReplicas": 1``, ``"ChannelIndexShards": 1``, ``"UserIndexReplicas": 1``, and ``"UserIndexShards": 1``, to enable autocompletion for Elasticsearch queries.

### API Changes
 - Added ``GetUsers`` API method to add the ability to list users.
 - Added the ``SearchPostsInTeam`` method to the plugin API to add the ability to search posts in a team.
 - Added ``GetTeamMembersForUser`` and ``GetChannelMembersForUser`` to the plugin API to add the ability to get team and channel members for a specific user.
 - Added ``GetBundleInfo() string`` method to the plugin API to add the ability to store assets elsewhere.
 
### Websocket Event Changes

> ### Database Changes

Reviewer 3: added bots roles/permissions, seel: https://github.com/mattermost/mattermost-server/pull/10507/files

Reviewer: Propose

```
Granted the following permissions for the System Admin, in preparation for an upcoming release of bot accounts:
 - PERMISSION_CREATE_BOT
 - PERMISSION_READ_BOTS
 - PERMISSION_READ_OTHERS_BOTS
 - PERMISSION_MANAGE_BOTS
 - PERMISSION_MANAGE_OTHERS_BOTS
```

   - Added ``"MinimumHashtagLength": 3``, to add the ability to change the minimum length of hashtags.

### API Changes
 - Added ``GetUsers`` API method to add the ability to list users.
 - Added the ``SearchPostsInTeam`` method to the plugin API to add the ability to search posts in a team.
 - Added ``GetTeamMembersForUser`` and ``GetChannelMembersForUser`` to the plugin API to add the ability to get team and channel members for a specific user.
 - Added ``GetBundleInfo() string`` method to the plugin API to add the ability to store assets elsewhere.

### Database Changes
 - Granted the following permissions for the System Admin, in preparation for an upcoming bot accounts feature:
   - PERMISSION_CREATE_BOT
   - PERMISSION_READ_BOTS
   - PERMISSION_READ_OTHERS_BOTS
   - PERMISSION_MANAGE_BOTS
   - PERMISSION_MANAGE_OTHERS_BOTS

### Known Issues
 - On a server using a subpath, the URL opens a blank page if the system admin changes the Site URL in the System Console UI. The system admin should restart the server to fix it.
 - Login does not work when Custom Terms of Service is enabled and MFA is enforced.
 - Google login fails on the Classic mobile apps.

>  - User can receive a video call from another browser tab while already on a call.

Reviewer 2: To what is this referring?

Reviewer 2: Specifically:
> - User can receive a video call from another browser tab while already on a call.

Author: I wasn't able to find a Jira ticket so I removed it from the list for now,

Reviewer: PS: This was an issue with the legacy WebRTC calls
